scripts/tutorial-uploader/lib/api.test.ts

```diff
@@ -0,0 +1,171 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { randomBytes } from "crypto";
+
+import { describe, expect, test } from "@jest/globals";
+import { readItem } from "@directus/sdk";
+
+import { API } from "./api";
+import { type LocalTutorialData } from "./local-tutorial-data";
+
+/* Skip tests if environment is not set */
+const maybeDescribe =
+  process.env.JEST_TEST_TUTORIAL_UPLOADER === "true" ? describe : describe.skip;
+
+/* Create test data */
+const createdSlugs: string[] = []; // To teardown afterwards
+function generateTutorialData(): LocalTutorialData {
```

Comment:
What if instead, you have a single variable inside the `describe` like `MOCK_TUTORIAL`? You can declare the object `as const` to ensure that one test doesn't overwrite the object and mess up other tests. Rather than `afterAll`, you'd use `afterEach` to remove `MOCK_TUTORIAL.slug`.

I think this might make things more clear. Right now, it's not very idiomatic to have `createdSlugs` and `generateTutorialData` outside the `desribe()`, and it took me a minute to convince myself that it's safe to use `afterAll` rather than `afterEach` because the tutorial slug will be unique.

--

If you take this recommendation, you don't need to use a random test ID. You can call it something like `test-tutorial`. 

Note that cleanup does not always happen properly, like if someone uses ctrl-c to exit out of the script. So, you might want to even delete this test entry in `beforeAll`.

Reply:
Ok nice, I assumed the jest tests ran in parallel so I added the IDs to avoid conflicts. Fixed in ad843ecec.